src/components/ThumbsFeedback/ThumbsFeedback.tsx

```diff
@@ -0,0 +1,165 @@
+
+const MAX_COMMENT_LENGTH: number = 100;
+
+const ThumbsFeedback = (): JSX.Element => {
+  const [feedback, setFeedback] = useState<string | null>(null);
+  const [comment, setComment] = useState<string>("");
+  const [showButtons, setShowButtons] = useState<boolean>(true);
+  const [isSubmitted, setIsSubmitted] = useState<boolean>(false);
+  const [isSidebarScrollable, setIsSidebarScrollable] = useState<boolean>(true);
+  const location = useLocation();
+
+  useEffect(() => {
+    setShowButtons(true);
+    setIsSubmitted(false);
+    setFeedback(null);
+    setComment("");
+  }, [location.pathname]);
+
+  // Currently feedback will be hidden if sidebar is scrollable AND within 350 of the bottom as this is the height when the textarea is active
+  // Can plan to implement this better going forward
+  useEffect(() => {
+    const checkSidebarScrollable = (): void => {
+      const tocDesktop: Element | null = document.querySelector('.theme-doc-toc-desktop');
+      if (tocDesktop) {
+        const isScrollable: boolean = tocDesktop.scrollHeight > tocDesktop.clientHeight;
+        const viewportHeight: number = window.innerHeight;
+        const tocRect = tocDesktop.getBoundingClientRect();
+        const spaceBelow: number = viewportHeight - tocRect.bottom;
+        const wouldOverflowWithTextArea: boolean = spaceBelow < 300;
+
+        setIsSidebarScrollable(isScrollable || wouldOverflowWithTextArea);
+
+      }
+    };
+    checkSidebarScrollable();
+
+    window.addEventListener('resize', checkSidebarScrollable);
+    const timer = setTimeout(checkSidebarScrollable, 100);
+
+    return (): void => {
+      window.removeEventListener('resize', checkSidebarScrollable);
+      clearTimeout(timer);
+    };
+  }, [location.pathname]);
+
+  const handleFeedbackClick = async (feedbackValue: string): Promise<void> => {
+    setFeedback(feedbackValue);
+    setShowButtons(false);
+
+    trackEvent({
+      event_name: 'feedback_thumb_click',
+      event_category: 'feedback',
+      event_label: `thumbs_${feedbackValue}`,
+      custom_parameters: {
+        thumb_direction: feedbackValue
+      }
+    });
+  };
+
+  const handleSubmit = async (event: FormEvent<HTMLFormElement>): Promise<void> => {
+    event.preventDefault();
+
+    if (!isValidCommentLength(comment, MAX_COMMENT_LENGTH)) {
+      return;
+    }
+
+    const trimmedComment = comment.trim();
+
+    if (!containsPII(trimmedComment)) {
+      trackEvent({
+        event_name: 'feedback_comment_submit',
+        event_category: 'feedback',
+        event_label: `comment_thumbs_${feedback}`,
+        custom_parameters: {
+          comment_text: trimmedComment,
+          thumb_direction: feedback
+        }
+      });
+    }
+
+    setIsSubmitted(true);
+  };
+
+  if (isSubmitted) {
+    return <p>Thank you for your feedback.</p>;
+  }
+
+  const shouldShowFeedback: boolean = !isSidebarScrollable;
+
+  if (!shouldShowFeedback) {
+    return null;
+  }
```

Comment:
Would it make sense to set the `if` condition to `!isSidebarScrollable` instead of making a new declaration?